sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/persistence/DatabasePersistenceAndroidTest.java

```diff
@@ -636,17 +628,104 @@ private void getAllLogs(DatabasePersistence persistence, int numberOfLogs, int s
         int expected = 0;
         do {
             numberOfLogs -= expected;
-            persistence.getLogs("test", sizeForGetLogs, outputLogs);
+            persistence.getLogs("test", Collections.<String>emptyList(), sizeForGetLogs, outputLogs);
             expected = Math.min(Math.max(numberOfLogs, 0), sizeForGetLogs);
             assertEquals(expected, outputLogs.size());
             outputLogs.clear();
         } while (numberOfLogs > 0);
 
         /* Get should be 0 now. */
-        persistence.getLogs("test", sizeForGetLogs, outputLogs);
+        persistence.getLogs("test", Collections.<String>emptyList(), sizeForGetLogs, outputLogs);
         assertEquals(0, outputLogs.size());
     }
 
+    @Test
+    public void getLogsFilteringOutDisabledKeys() throws PersistenceException {
+
+        /* Initialize database persistence. */
+        DatabasePersistence persistence = new DatabasePersistence(sContext);
+
+        /* Set a mock log serializer. */
+        LogSerializer logSerializer = new DefaultLogSerializer();
+        logSerializer.addLogFactory(MockCommonSchemaLog.TYPE, new MockCommonSchemaLogFactory());
+        persistence.setLogSerializer(logSerializer);
+        try {
+
+            /* Test constants. */
+            int numberOfLogsPerKey = 10;
+
+            /* Generate and persist some logs with a first iKey. */
```

Comment:
Could we put this in a utility class?  Or maybe just a private method?

Reply:
What part do you want in a private method? Not sure to understand as it's used only once.

Reply:
I suppose that's fine.  It's basically these next 10 lines are repeated three times in this method with minimal changes.  It could be `generatePersistedLogs(String key, int numLogs)`

Reply:
ah github was showing the constant, ok.

Reply:
9cacdcaea279e74476dc0fb44d92055f9581e89e addresses it